streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/watcher/FlinkJobStatusWatcher.scala

```diff
@@ -276,7 +277,9 @@ class FlinkJobStatusWatcher(conf: JobStatusWatcherConfig = JobStatusWatcherConfi
           }
         } else if (isConnection) {
           logger.info("The deployment is deleted and enters the task failure process.")
-          FlinkJobState.FAILED
+          val jobId = trackId.jobId
+          val archivePath = trackController.flinkArchives.get(jobId)
+          FlinkJobState.of(FetchArchives.fetchArchives(trackId.jobId, archivePath))
```

Comment:
1. Should we clear the jobId in the `flinkArchives` after get the jobState?
2. If the StreamPark backend service is restarted. how to get the archivePath?

Reply:
1.Yes, you should clear the cache
2.The state of the restart is guaranteed by the state field in the t_flink_app table in mysql

Reply:
`val archivePath = trackController.flinkArchives.get(jobId)`

We get `archivePath` from memory cache, I didn't see the `archivePath` store in mysql. 

Please correct me if I'm wrong.